### PR TITLE
Complementary areas use isActiveByDefault prop and local storage to determine whether is active or not

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
## Description
This PR removes static defaults setting which complementary areas are active (sidebars). Instead of static defaults, it proposes a flag "isActiveByDefault" that, when true, makes a complementary area active by default on the first-page load. On follow up page loads, the setting is retrieved from local storage.

cc: @youknowriad, @gziolo 

## How has this been tested?
I used `window.localStorage.clear();`to clean the local storage.
I verified on edit-post and edit-site the default state of sidebars is kept as before.
I changed sidebars and verified their state is still persisted across page loads.

### CODE REVIEW

File: packages/edit-site/src/components/sidebar/index.js [hunk 1]
```diff
@@ -5,10 +5,16 @@ import { createSlotFill } from '@wordpress/components';
 import { ComplementaryArea } from '@wordpress/interface';
 import { __ } from '@wordpress/i18n';
 import { cog, pencil } from '@wordpress/icons';
+import { Platform } from '@wordpress/element';
 
 const { Slot: InspectorSlot, Fill: InspectorFill } = createSlotFill(
 	'EditSiteSidebarInspector'
 );
+
+const BLOCK_INSPECTOR_ACTIVE_BY_DEFAULT = Platform.select( {
```
Comment (youknowriad) [inline packages/edit-site/src/components/sidebar/index.js:14]: Is this default value different  between native and web or between web and web mobile?

Comment (jorgefilipecosta) [reply packages/edit-site/src/components/sidebar/index.js:14]: The difference is between native and web. We had a /packages/interface/src/store/defaults.native.js file specifying the default for native is to contain no open sidebars.

**Comment by jorgefilipecosta** [comment]
This PR was rebased and should be ready for a final look.

**Review by youknowriad — APPROVED** [review]
Thanks Jorge

**Pull request merged**